src/main/java/com/nicusa/controller/DrugController.java

```diff
@@ -73,6 +75,70 @@
     return new ResponseEntity<>(drugAssembler.toResource(drug), HttpStatus.OK);
   }
 
+  @Transactional
+  @ResponseBody
+  @RequestMapping(value = "/portfolio/drug", method = RequestMethod.POST, consumes = "application/json")
+  public ResponseEntity<?> saveDrug(@AuthenticationPrincipal Principal principal,
```

Comment:
We should either use the principal in the method call or the call the securityController.getAuthenticatedUserProfileId() we don't need both remove one.